ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java

```diff
@@ -1246,4 +1253,322 @@ private void applyDelete(MessageInfo messageInfo, ReplicateBlobRequest replicate
       }
     }
   }
+
+  /**
+   * Used to update common metrics for the incoming request.
+   */
+  public class RequestMetricsUpdater implements RequestVisitor {
```

Comment:
I don't think we need to use visitor pattern given that we already have lots of code branches that do a switch statement based on RequestType. It seems like the handleXXXRequest is more suitable to be converted to a visitor. But, anyway, it's fine to use it.

Reply:
+1, these visit methods are just adding more unnecessary code